geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/ListGatewayCommand.java

```diff
@@ -53,9 +57,23 @@ public ResultModel listGateway(
           help = CliStrings.LIST_GATEWAY__MEMBER__HELP) String[] onMember,
       @CliOption(key = {CliStrings.GROUP, CliStrings.GROUPS},
           optionContext = ConverterHint.MEMBERGROUP,
-          help = CliStrings.LIST_GATEWAY__GROUP__HELP) String[] onGroup)
+          help = CliStrings.LIST_GATEWAY__GROUP__HELP) String[] onGroup,
+      @CliOption(key = {CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY},
+          specifiedDefaultValue = "true",
+          help = CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY__HELP) Boolean showReceiversOnly,
+      @CliOption(key = {CliStrings.LIST_GATEWAY__SHOW_SENDERS_ONLY},
+          specifiedDefaultValue = "true",
+          help = CliStrings.LIST_GATEWAY__SHOW_SENDERS_ONLY__HELP) Boolean showSendersOnly)
+
       throws Exception {
 
+    if (showReceiversOnly == null) {
+      showReceiversOnly = false;
+    }
+    if (showSendersOnly == null) {
+      showSendersOnly = false;
+    }
```

Comment:
If I set them to "false" I cannot differentiate when they are false because the command was executed with that option from when they have not been introduced. I check the null value in the interceptor.

Reply:
You're right, having an `unspecifiedDefaultValue` set means you can't check if the user has specified both options in the interceptor.